src/dir.rs

```diff
@@ -1,15 +1,25 @@
+#[cfg(not(target_os = "redox"))]
```

Comment:
It looks like the only thing in this file that's still defined on Redox is `Entry`.  But what good is `Entry` without the other parts?  I think you should revert all changes to this file, and simply cfg out the entire module from src/lib.rs.

Reply:
You are right, changed it.